DOCS DOCS DOCS (#34)
fhdsl/metricminer: README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

> # metricminer

Reviewer: woohoo! looks good - maybe we should link to our article about metrics? [arXiv:2306.03255](https://arxiv.org/abs/2306.03255)

> # metricminer

Author: Yes yes! For sure! I'll add

`metricminer` is an R package that helps you mine metrics on common places on the web through the power of their APIs.

> It also helps make the data in a format that is easily used for a dashboard or other purposes.

Reviewer: ```suggestion
It also helps format the data so that it can easily be used for a dashboard or other purposes.
```

Reviewer: just simplifying this

> It also helps make the data in a format that is easily used for a dashboard or other purposes.

Author: ```suggestion
It also helps make the data in a format that is easily used for a dashboard or other purposes.
You can read more about [suggested techniques for metrics in our associated manuscript](https://arxiv.org/abs/2306.03255).
```

It will have an associated [dashboard template](https://github.com/fhdsl/metricminer-dashboard) and tutorials to help you fully use the data you retrieve with `metricminer`  (but these are still under development!)

> You can [read the metricminer package documentation here](https://hutchdatascience.org/metricminer/).

Author: ```suggestion
- You can [read the metricminer package documentation here](https://hutchdatascience.org/metricminer/).
- And you can read more about metric collection in our [associated manuscript -- currently a preprint](https://arxiv.org/abs/2306.03255). 
```

## Apps supported

Currently `metricminer` supports mining data from:

- [Calendly](https://calendly.com/)
- [GitHub](https://github.com/)
- [Google Analytics](https://analytics.google.com/analytics/academy/course/6)
- [Google Forms](https://www.google.com/forms/about/)
- [Slido](https://admin.sli.do/events) export files stored on Googledrive

## Data format options

> `metricminer` attempts to retrieve API data for you and give you it to you in a format that is a tidy data.frame.

Reviewer: ```suggestion
`metricminer`  retrieves API data for you and gives it to you in a format that is a tidy data.frame.
```

> this means metricminer has to be opinionated about what metrics it returns so it fits in a useful and human ready to read data frame.

Reviewer: ```suggestion
this means metricminer has to be opinionated about what metrics it returns so it fits in a useful data frame easily read by humans.
```

If you find that the data returned is not what you need you have two options (these options can be pursued concurrently):

1. You can set the `dataformat` argument to `"raw"` to see the original, unedited JSON formatted data as it was returned from the API. Then you can personally look for the data that you want and extract it.

> 2. You can post a GitHub issue to explain why the metric missing from the data frame formatted data should be included. And if possible and reasonable, we can work on including that data in the next version of `metricminer`.

Reviewer: ```suggestion
2. You can post a GitHub issue to explain why the metric missing from the data frame of formatted data should be included. And if possible and reasonable, we can work on including that data in the next version of `metricminer`.
```

```

### Calendly

You can retrieve calendly events information using this type of workflow:
```
authorize("calendly")
user <- get_calendly_user()
events <- list_calendly_events(user = user$resource$uri)
```

### Google Analytics

You can retrieve Google Analytics data for websites like this.

First you have to retrieve your account information after you've authorized.

```
authorize("google")
accounts <- get_ga_user()
```

Then you need to retrieve the properties (aka usually the websites you are tracking)
underneath that account.

```
properties_list <- get_ga_properties(account_id = accounts$id[1])
```

Just need to shave off the `properties/` bit from this string.

```
property_id <- gsub("properties/", "", properties_list$properties$name[1])
```

Now we can collect some stats.

In Google Analytics `metrics` are your basic numbers (how many visits to your website, etc.).
```
metrics <- get_ga_stats(property_id, stats_type = "metrics")
```
Whereas `dimensions` are more  a list of events that have happened. So here's a list of people that have logged on.
```
dimensions <- get_ga_stats(property_id, stats_type = "dimensions")
```

> Lastly, we have a third option of collecting `link_clicks` and the links they have clicked. This is also known as a dimension according to Google analytics, but often it isn't compatible for us to download link click data at the same time as other dimension data so in `metricminer` we collect them separately.

Reviewer: ```suggestion
Lastly, we have a third option of collecting `link_clicks` and the links they have clicked. This is also known as a dimension according to Google Analytics, but often it isn't compatible for us to download link click data at the same time as other dimension data so in `metricminer` we collect them separately.
```

```
link_clicks <- get_ga_stats(property_id, stats_type = "link_clicks")
```

### Google Forms

You can retrieve Google form information and responses like this:
```
authorize("google")
form_url <- "https://docs.google.com/forms/d/1Z-lMMdUyubUqIvaSXeDu1tlB7_QpNTzOk3kfzjP2Uuo/edit"
form_info <- get_google_form(form_url)
```

### Slido

If you have used Slido for interactive slide sessions and collected that info and exported it to your googledrive you can use `metricminer` to collect that data as well.

```
drive_id <- "https://drive.google.com/drive/folders/0AJb5Zemj0AAkUk9PVA"
slido_data <- get_slido_files(drive_id)
```

## Bulk Retrievals

> Maybe you just want to retrieval it ALL. We have som wrapper functions that will attempt to do this for you.

Reviewer: ```suggestion
Maybe you just want to retrieve it ALL. We have some wrapper functions that will attempt to do this for you.
```
